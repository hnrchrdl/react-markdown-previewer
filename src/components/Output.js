import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import ReactMarkdown from 'react-markdown';

const OutputContainer = styled.div`
  width: 100%;
  padding: 20px;
  box-sizing: border-box;
  background: #eee;
`;

const Output = props => (
  <OutputContainer id="preview">
    <ReactMarkdown source={props.input} />
  </OutputContainer>
);

const mapStateToProps = state => ({
  input: state.io.input
});

export default connect(mapStateToProps)(Output);
