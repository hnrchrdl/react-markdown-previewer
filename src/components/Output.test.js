import React from 'react';
import Output from './Output';
import renderer from 'react-test-renderer';
import store from '../store';

test('App renders and matches snapshot', () => {
  const component = renderer.create(<Output store={store} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
