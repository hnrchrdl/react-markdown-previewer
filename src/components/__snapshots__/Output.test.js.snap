// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders and matches snapshot 1`] = `
<div
  className="sc-bdVaJa dECICN"
  id="preview"
>
  <div>
    <h1>
      a header (H1 size)
    </h1>
    <h2>
      a sub header (H2 size)
    </h2>
    <p>
        
      <a
        href=""
      >
        a link
      </a>
      
        
      <code>
        inline code
      </code>
      <br />
        
      <code>
        a code block
      </code>
      <br />
        1. a list item
    </p>
    <blockquote>
      <p>
        a blockquote
        <br />
          
        <img
          alt="an image"
          src="https://cdn.vox-cdn.com/thumbor/YqrPXWiAFv9TuM0tUi_i1pZEPyE=/0x0:512x512/920x0/filters:focal(0x0:512x512):format(webp)/cdn.vox-cdn.com/uploads/chorus_asset/file/6808705/Cat_101.0.png"
        />
        
          
        <strong>
          and bolded text
        </strong>
      </p>
    </blockquote>
  </div>
</div>
`;
